cache/test/cache/s3_download_worker_test.exs

```diff
@@ -0,0 +1,167 @@
+defmodule Cache.Workers.S3DownloadWorkerTest do
+  use ExUnit.Case, async: false
+  use Mimic
+
+  import ExUnit.CaptureLog
+
+  alias Cache.Repo
+  alias Cache.S3DownloadWorker
+  alias Ecto.Adapters.SQL.Sandbox
+
+  setup do
+    :ok = Sandbox.checkout(Repo)
+    Sandbox.mode(Repo, {:shared, self()})
+
+    :ok
+  end
+
+  describe "enqueue_download/3" do
+    test "enqueues download job" do
+      account_handle = "test_account"
+      project_handle = "test_project"
+      id = "test_hash"
+
+      expect(Oban, :insert, fn changeset ->
+        {:ok,
+         %{
+           changeset
+           | changes:
+               Map.put(changeset.changes, :args, %{
+                 "key" => "test_account/test_project/cas/test_hash",
+                 "account_handle" => "test_account",
+                 "project_handle" => "test_project",
+                 "id" => "test_hash"
+               })
+         }}
+      end)
+
+      capture_log(fn ->
+        {:ok, changeset} = S3DownloadWorker.enqueue_download(account_handle, project_handle, id)
+
+        assert changeset.changes.args == %{
+                 "key" => "test_account/test_project/cas/test_hash",
+                 "account_handle" => "test_account",
+                 "project_handle" => "test_project",
+                 "id" => "test_hash"
+               }
+
+        assert changeset.changes.worker == "Cache.S3DownloadWorker"
+      end)
+    end
+  end
+
+  describe "perform/1" do
+    test "successfully downloads file from S3 when it exists" do
+      key = "test_account/test_project/cas/test_hash"
+      account_handle = "test_account"
+      project_handle = "test_project"
+      id = "test_hash"
+      {:ok, tmp_dir} = Briefly.create(directory: true)
+      local_path = Path.join(tmp_dir, "test_hash")
+
+      expect(Cache.S3, :exists?, fn ^key -> true end)
+      expect(Cache.Disk, :artifact_path, fn ^key -> local_path end)
+
+      expect(ExAws.S3, :download_file, fn "test-bucket", ^key, ^local_path ->
+        {:download_operation, "test-bucket", key, local_path}
+      end)
+
+      expect(ExAws, :request, fn _download ->
```

Comment:
I'd match this one against the download request that you've stubbed in line 66 to make sure the right arguments are passed to `request`.